execution.go

```
@@ -172,6 +182,21 @@
 	return txs, nil
 }
 
+func (c *EngineAPIExecutionClient) getBlockHash(ctx context.Context, height uint64) (common.Hash, error) {
+	var block map[string]interface{}
+	err := c.engineClient.CallContext(
+		ctx, &block, "eth_getBlockByNumber", rpc.BlockNumber(height), false)
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int64

(gosec)